### PR TITLE
Wishlist management

### PR DESCRIPTION
**Pull request by Balakumar3018**
Wishlist functionality added.
1. Can add wishlist from product list and remove from there
2. Can add wishlist from cart page and navigate from ther
3. Can remove wishlist from wishlist page

Yet to add responsiveness

Preview - https://timely-pony-59f08f.netlify.app/

Please give suggestions

### CODE REVIEW

File: src/components/navigation-top/navigation-top.jsx [hunk 1]
```diff
@@ -19,11 +21,11 @@ export function NavigationTop(){
                     </div>
                     <Link to="/wishlist" className="heart-icon">
                         <i className="fa-solid fa-heart"></i>
-                        <span className="heart-badge"></span>
+                        <span className="heart-badge">{wishlistState.wishlist.length >0 ? wishlistState.wishlist.length : ''}</span>
```
Comment (sashaboi) [inline src/components/navigation-top/navigation-top.jsx:24]: Could you try to have a badge where there is a round background and a number in between ? 
would be much easier to look at , currently the colors merge together creating a less than perfect UI experience.

File: src/components/Product-card/ProductCard.jsx [hunk 2]
```diff
@@ -28,8 +29,19 @@ export const ProductCard=({product})=>{
                             Add to cart
                         </button>
                     ) }
-                    
-                    <button className="btn btn-secondary">Add to wishlist</button>
+                    {
+                        wishlistState.wishlist.find((item) => item._id === _id) ? (
+                                <button className="btn btn-secondary"
+                                onClick={()=>wishlistDispatch({type:"RemoveFromWishlist",payload:_id})}>
```
Comment (sashaboi) [inline src/components/Product-card/ProductCard.jsx:35]: could some animation be done for on hover of just the button? 
also the buttons section and the top image /title part could use a bit of a gap in my opinion.

**Review by sashaboi — APPROVED** [review]
The functionality is nothing less than perfect! 

A few suggestions : 
You could write a remove from cart function on the decrement quantity button when quantity is zero?
A few css suggestions made, they are non essential , but hope they get considered in future versions.

**Review by teja538 — APPROVED** [review]
suggestions:
- use some margin top and bottom for the buttons in the product card for good Ux
- in the product card it will be better if you add Rs or rupee icon for price and star icon for rating
- logic looks good in the code

**Pull request merged**